### ISSUE TITLE
.NET Core 2.1

### ISSUE THREAD

**Issue by leecow** [dotnet/core#1614]
# .NET Core 2.1.0

[Blog post](https://blogs.msdn.microsoft.com/dotnet/2018/05/30/announcing-net-core-2-1/)
[Release Notes](https://github.com/dotnet/core/blob/master/release-notes/2.1/2.1.0.md)

Please report any issues you find with 2.1, either responding to this issue, creating a new issue or creating a new issue in one of the following repos:

* [aspnet/Home](https://github.com/aspnet/Home) - for ASP.NET Core questions
* [dotnet/cli](https://github.com/dotnet/cli/issues) - for CLI tools and questions
* [dotnet/corefx](https://github.com/dotnet/corefx/issues) - for API issues and questions
* [dotnet/coreclr](https://github.com/dotnet/coreclr/issues) - for runtime issues
* [dotnet/core-setup](https://github.com/dotnet/core-setup/issues) - for installer issues
* [nuget/home](https://github.com/nuget/home/issues) - for NuGet questions and issues

**Comment by uffebjorklund** [comment]
@leecow getting a 404 on the [link](https://blogs.msdn.microsoft.com/dotnet/2018/05/30/announcing-net-core-2-1/) to the blog post. Thinking that it might be correct, but just not published yet?

**Comment by winzig** [comment]
Link to "Known Issues" from [here](https://github.com/dotnet/core/blob/master/release-notes/2.1/2.1.0.md) is also returning 404.

**Comment by leecow** [comment]
@uffebjorklund  and @winzig - both should be up now.

**Comment by jnm2** [comment]
Link to https://github.com/dotnet/core/blob/master/release-notes/2.1/api-diff/rc1/2.1.md is broken in https://github.com/dotnet/core/blob/master/release-notes/2.1/2.1.0.md#api-changes

Looks like it should have been https://github.com/dotnet/core/blob/master/release-notes/2.1/api-diff/2.0-vs-2.1.md.

**Comment by leecow** [comment]
@jnm2 - fixed!

**Comment by annymsMthd** [comment]
The ubuntu packages for bionic seem to be missing "dotnet-sdk-2.1.300" and reverts to installing the preview. It looks like the package "dotnet-sdk-2.1" is pointing to the correct deb file though.

**Comment by leecow** [comment]
@annymsMthd - We need to make an update to the install instructions so to refer to `dotnet-sdk-2.1`. This is part of the update to make the installers behave themselves for updates.

**Comment by rowanmiller** [comment]
Site is updated - https://www.microsoft.com/net/download/linux-package-manager/rhel/sdk-current

**Comment by damageboy** [comment]
I have a weird issue with the trusty repo (perhaps other ubuntu repos are also affected)..

I'm using a mirroring too to mirror the entire repo for installation inside a segragated network consisting of hundreds of machines.

Unfortunately, one of packages in the repo seems to have a bad hash value in the Packages file...

While the package itself (addlogin) is not a dependency of dotnet-sdk-2.1.300 in itself, the mirroring tool has no way of knowing this, and fails on the hash mismatch regardless...

Here's is the information:

The 2.1.200 SDK depends on "aadlogin-selinux_1.0.004870001_amd64.deb" but the hash mismatches

# General

When I do:
```
wget https://packages.microsoft.com/repos/microsoft-ubuntu-trusty-prod/pool/main/a/aadlogin-selinux/aadlogin-selinux_1.0.004870001_amd64.deb
--2018-05-10 00:16:04--  https://packages.microsoft.com/repos/microsoft-ubuntu-trusty-prod/pool/main/a/aadlogin-selinux/aadlogin-selinux_1.0.004870001_amd64.deb
Resolving packages.microsoft.com (packages.microsoft.com)... 13.80.10.205
Connecting to packages.microsoft.com (packages.microsoft.com)|13.80.10.205|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 13378 (13K) [application/octet-stream]
Saving to: ‘aadlogin-selinux_1.0.004870001_amd64.deb’

100%[=================================================================================================================================================================>] 13,378      --.-K/s   in 0s

2018-05-10 00:16:05 (265 MB/s) - ‘aadlogin-selinux_1.0.004870001_amd64.deb’ saved [13378/13378]

$ md5sum aadlogin-selinux_1.0.004870001_amd64.deb
1341006af46b5b9f7ec7a4458e96ee5d  aadlogin-selinux_1.0.004870001_amd64.deb
$ sha1sum aadlogin-selinux_1.0.004870001_amd64.deb
9e0a0a27500889a18a3b52fe728a377266ece884  aadlogin-selinux_1.0.004870001_amd64.deb
```

If you download the package release file:
```
https://packages.microsoft.com/repos/microsoft-ubuntu-trusty-prod/dists/trusty/main/binary-amd64/Packages.bz2
```

The package is signed with the following hashes:
```
Package: aadlogin
Priority: extra
Section: default
Installed-Size: 368
Maintainer: Azure AD <azuread@microsoft.com>
Architecture: amd64
Version: 1.0.004880001
Depends: libcurl3, uuid-runtime
Filename: pool/main/a/aadlogin/aadlogin_1.0.004880001_amd64.deb
Size: 136322
MD5sum: f38961a87872d754fcf9beeb6cea83c6
SHA1: c05eb728d1208b5edbb8c07b3f3763a76da842ed
SHA256: c9e1819732b5da3ba984abf8fe9fbfd38637f3569fb726268c0b714f933f1934
SHA512: 5da10a913581539a4f0ce90668624676316864f9624faeaae289bb4e6bc49f1f911161fd5b7b2f993db7b6c337b049bd2c78fec72bd5026e8371bdf95c6847d7
Description: AAD NSS and PAM extensions
Homepage: https://azure.com
License: MIT
Vendor: Microsoft
```

I'm guessing something got messed up?

**Comment by leecow** [comment]
@damageboy - I'll check with the product repo owners with respect to the hash. 

Where are you seeing that the SDK has a dependency on aadlogin? The only defined dependencies are on aspnetcore-runtime and dotnet-runtime and it's not a dependency through any of the other packages.

**Comment by tanaka-takayoshi** [comment]
While installing .NET Core 2.1 in Fedora 27, I can see some dependent packages are RC version. Is it expected?

~~~
$ sudo dnf rm dotnet-*
$ sudo rm /etc/yum.repos.d/microsoft.repo
$ sudo rpm --import https://packages.microsoft.com/keys/microsoft.asc
$ wget -q https://packages.microsoft.com/config/fedora/27/prod.repo
$ sudo mv prod.repo /etc/yum.repos.d/microsoft-prod.repo
$ sudo dnf update
packages-microsoft-com-prod                                                                                                                                                        162 kB/s |  55 kB     00:00    
Last metadata expiration check: 0:00:00 ago on Thu 31 May 2018 09:48:57 AM JST.
Dependencies resolved.
Nothing to do.
Complete!
$ sudo dnf install dotnet-sdk-2.1
Last metadata expiration check: 0:00:11 ago on Thu 31 May 2018 09:48:57 AM JST.
Dependencies resolved.
===================================================================================================================================================================================================================
 Package                                                      Arch                                  Version                                       Repository                                                  Size
===================================================================================================================================================================================================================
Installing:
 dotnet-sdk-2.1                                               x86_64                                2.1.300-1                                     packages-microsoft-com-prod                                109 M
Installing dependencies:
 aspnetcore-runtime-2.1                                       x86_64                                2.1.0-1                                       packages-microsoft-com-prod                                 29 M
 dotnet-host                                                  x86_64                                2.1.0_rc1-1                                   packages-microsoft-com-prod                                 43 k
 dotnet-hostfxr-2.1                                           x86_64                                2.1.0-1                                       packages-microsoft-com-prod                                195 k
 dotnet-runtime-2.1                                           x86_64                                2.1.0-1                                       packages-microsoft-com-prod                                 27 M
 dotnet-runtime-deps-2.1                                      x86_64                                2.1.0-1                                       packages-microsoft-com-prod                                2.8 k
 dotnet-runtime-deps-2.1.0-rc1                                x86_64                                2.1.0_rc1-1                                   packages-microsoft-com-prod                                2.8 k

Transaction Summary
===================================================================================================================================================================================================================
Install  7 Packages

Total download size: 165 M
Installed size: 275 M
Is this ok [y/N]: y
Downloading Packages:
(1/7): dotnet-runtime-2.1.0-x64.rpm                                                                                                                                                3.3 MB/s |  27 MB     00:08    
(2/7): dotnet-hostfxr-2.1.0-x64.rpm                                                                                                                                                905 kB/s | 195 kB     00:00    
(3/7): dotnet-runtime-deps-2.1.0-fedora.27-x64.rpm                                                                                                                                  32 kB/s | 2.8 kB     00:00    
(4/7): dotnet-host-2.1.0-rc1-x64.rpm                                                                                                                                               255 kB/s |  43 kB     00:00    
(5/7): dotnet-runtime-deps-2.1.0-rc1-fedora.27-x64.rpm                                                                                                                              38 kB/s | 2.8 kB     00:00    
(6/7): aspnetcore-runtime-2.1.0-x64.rpm                                                                                                                                            1.9 MB/s |  29 MB     00:15    
(7/7): dotnet-sdk-2.1.300-x64.rpm                                                                                                                                                  2.9 MB/s | 109 MB     00:37    
-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
Total                                                                                                                                                                              4.4 MB/s | 165 MB     00:37     
Running transaction check
Transaction check succeeded.
Running transaction test
Transaction test succeeded.
Running transaction
  Preparing        :                                                                                                                                                                                           1/1 
  Installing       : dotnet-runtime-deps-2.1.0-rc1-2.1.0_rc1-1.x86_64                                                                                                                                          1/7 
  Installing       : dotnet-host-2.1.0_rc1-1.x86_64                                                                                                                                                            2/7 
  Running scriptlet: dotnet-host-2.1.0_rc1-1.x86_64                                                                                                                                                            2/7 
Creating dotnet host symbolic link: /usr/bin/dotnet
  Installing       : dotnet-hostfxr-2.1-2.1.0-1.x86_64                                                                                                                                                         3/7 
  Installing       : dotnet-runtime-deps-2.1-2.1.0-1.x86_64                                                                                                                                                    4/7 
  Installing       : dotnet-runtime-2.1-2.1.0-1.x86_64                                                                                                                                                         5/7 
  Installing       : aspnetcore-runtime-2.1-2.1.0-1.x86_64                                                                                                                                                     6/7 
  Installing       : dotnet-sdk-2.1-2.1.300-1.x86_64                                                                                                                                                           7/7 
  Running scriptlet: dotnet-sdk-2.1-2.1.300-1.x86_64                                                                                                                                                           7/7 
This software may collect information about you and your use of the software, and send that to Microsoft.
Please visit http://aka.ms/dotnet-cli-eula for more information.
Welcome to .NET Core!
---------------------
Learn more about .NET Core: https://aka.ms/dotnet-docs
Use 'dotnet --help' to see available commands or visit: https://aka.ms/dotnet-cli-docs

Telemetry
---------
The .NET Core tools collect usage data in order to help us improve your experience. The data is anonymous and doesn't include command-line arguments. The data is collected by Microsoft and shared with the community. You can opt-out of telemetry by setting the DOTNET_CLI_TELEMETRY_OPTOUT environment variable to '1' or 'true' using your favorite shell.

Read more about .NET Core CLI Tools telemetry: https://aka.ms/dotnet-cli-telemetry

Configuring...
--------------
A command is running to populate your local package cache to improve restore speed and enable offline access. This command takes up to one minute to complete and only runs once.
Running as unit: run-ra6a83a23f169471ea8955ff301397889.service
  Verifying        : dotnet-sdk-2.1-2.1.300-1.x86_64                                                                                                                                                           1/7 
  Verifying        : aspnetcore-runtime-2.1-2.1.0-1.x86_64                                                                                                                                                     2/7 
  Verifying        : dotnet-runtime-2.1-2.1.0-1.x86_64                                                                                                                                                         3/7 
  Verifying        : dotnet-hostfxr-2.1-2.1.0-1.x86_64                                                                                                                                                         4/7 
  Verifying        : dotnet-runtime-deps-2.1-2.1.0-1.x86_64                                                                                                                                                    5/7 
  Verifying        : dotnet-host-2.1.0_rc1-1.x86_64                                                                                                                                                            6/7 
  Verifying        : dotnet-runtime-deps-2.1.0-rc1-2.1.0_rc1-1.x86_64                                                                                                                                          7/7 

Installed:
  dotnet-sdk-2.1.x86_64 2.1.300-1            aspnetcore-runtime-2.1.x86_64 2.1.0-1                dotnet-host.x86_64 2.1.0_rc1-1     dotnet-hostfxr-2.1.x86_64 2.1.0-1     dotnet-runtime-2.1.x86_64 2.1.0-1    
  dotnet-runtime-deps-2.1.x86_64 2.1.0-1     dotnet-runtime-deps-2.1.0-rc1.x86_64 2.1.0_rc1-1    

[tatanaka@tatanaka-lp ~]$ ^C
[tatanaka@tatanaka-lp ~]$ 
Complete!
```

**Comment by lindexi** [comment]
Can we install it to Win7 without sp1?

**Comment by gortok** [comment]
The Known Issues page does not list a known issue we hit (and a pretty big one). The Alpine .NET Core images will segfault on a host running with SELinux on. https://github.com/dotnet/dotnet-docker/issues/561

**Comment by damageboy** [comment]
@leecow 
As for where I'm seeing this...
Again, this is a bit of a non-standard situation.
I'm seeing this while attempting to create a mirror of the repo... using an apt mirroring tool.

That is why even if there is NO dependency on addlogin through, for example, installing dotnet-sdk-2.1.300... The mirroring tool doesn't care or know about that. As far as a mirroring tool would be concerned it's all or nothing. And the moment it sees an hash mismatch on the Package release files, it's game over.

Again... Stressing that this is a less than standard situation, since what is failing is the apt repo MIRRORING rather than the installation.

I've previously opened an issue about this before, during the RC1 phase...
Here it is: https://github.com/dotnet/core/issues/1530

**Comment by leecow** [comment]
@tanaka-takayoshi - we're aware of the issue and tracking in https://github.com/dotnet/core/issues/1620

**Comment by leecow** [comment]
@damageboy - contacting the repo admins to see if they can resolve.

**Comment by JesseObrien** [comment]
Looks like .NET Core 2.1 is broken on Arch Linux.

I'm having the same issues as this person here: https://stackoverflow.com/questions/50689340/asp-net-core-2-1-on-arch-linux-does-not-run

I'm trying to run this project (simple project to just play with signalr 2.1 and such).

https://github.com/JesseObrien/signalr-jwt

Error:

```
It was not possible to find any compatible framework version
The specified framework 'Microsoft.AspNetCore.App', version '2.1.0' was not found.
  - Check application dependencies and target a framework version installed at:
      /opt/dotnet/
  - Installing .NET Core prerequisites might help resolve this problem:
      http://go.microsoft.com/fwlink/?LinkID=798306&clcid=0x409
  - The .NET Core framework and SDK can be installed from:
      https://aka.ms/dotnet-download
```

Same package installed, same `dotnet --info`:

```
NET Core SDK (reflecting any global.json):
 Version:   2.1.300
 Commit:    adab45bf0c

Runtime Environment:
 OS Name:     arch
 OS Version:
 OS Platform: Linux
 RID:         linux-x64
 Base Path:   /opt/dotnet/sdk/2.1.300/

Host (useful for support):
  Version: 2.1.0
  Commit:  caa7b7e2ba

.NET Core SDKs installed:
  2.1.300 [/opt/dotnet/sdk]
```

**Comment by Petermarcu** [comment]
You likely need to copy aspnet into the right location. Should work if you do that an console apps run fine.

**Comment by mrichards3** [comment]
This page doesn't have instructions or package links for .NET Core 2.1 in the "Install .NET Core for supported Red Hat Enterprise Linux (RHEL) versions" section:
https://docs.microsoft.com/en-us/dotnet/core/linux-prerequisites?tabs=netcore2x#install-net-core-for-supported-red-hat-enterprise-linux-rhel-versions

Are the RHEL packages still being finalized for .NET Core 2.1?

**Comment by KevinKibler** [comment]
It looks like the release candidate files were removed and replaced by the official release files. The yum package for dotnet-sdk-2.1.0 should be updated to download the new files.

```
> yum install dotnet-sdk-2.1
...
Downloading packages:
dotnet-host-2.1.0-rc1-x64.rpm  FAILED]  0.0 B/s |    0 B  --:--:-- ETA https://packages.microsoft.com/rhel/7/prod/dotnet-host-2.1.0-rc1-x64.rpm: [Errno 14] HTTPS Error 404 - Not Found
...
Error downloading packages:
  dotnet-host-2.1.0_rc1-1.x86_64: [Errno 256] No more mirrors to try.
  dotnet-runtime-deps-2.1.0-rc1-2.1.0_rc1-1.x86_64: [Errno 256] No more mirrors to try.
```

**Comment by tmds** [comment]
@mrichards3 rpms for RHEL should be available this week. We'll update those instructions once they are available.

**Comment by murrayju** [comment]
To add to what @KevinKibler said, it looks like the dependency tree still refers to some of the rc files that were deleted. You can reproduce this easily with docker:
```
docker run --rm centos:7 bash -c "rpm -Uvh https://packages.microsoft.com/config/rhel/7/packages-microsoft-prod.rpm && yum -y update && yum install -y https://packages.microsoft.com/rhel/7/prod/dotnet-sdk-2.1.300-x64.rpm"
```
Here is the relevant bit of output:
```
Examining /var/tmp/yum-root-lq2zV8/dotnet-sdk-2.1.300-x64.rpm: dotnet-sdk-2.1-2.1.300-1.x86_64
Marking /var/tmp/yum-root-lq2zV8/dotnet-sdk-2.1.300-x64.rpm to be installed
Resolving Dependencies
--> Running transaction check
---> Package dotnet-sdk-2.1.x86_64 0:2.1.300-1 will be installed
--> Processing Dependency: dotnet-runtime-2.1 >= 2.1.0 for package: dotnet-sdk-2.1-2.1.300-1.x86_64
Loading mirror speeds from cached hostfile
 * base: sjc.edge.kernel.org
 * extras: mirror.hostduplex.com
 * updates: centos-distro.cavecreek.net
--> Processing Dependency: aspnetcore-runtime-2.1 >= 2.1.0 for package: dotnet-sdk-2.1-2.1.300-1.x86_64
--> Running transaction check
---> Package aspnetcore-runtime-2.1.x86_64 0:2.1.0-1 will be installed
---> Package dotnet-runtime-2.1.x86_64 0:2.1.0-1 will be installed
--> Processing Dependency: dotnet-runtime-deps-2.1 >= 2.1.0 for package: dotnet-runtime-2.1-2.1.0-1.x86_64
--> Processing Dependency: dotnet-hostfxr-2.1 >= 2.1.0 for package: dotnet-runtime-2.1-2.1.0-1.x86_64
--> Running transaction check
---> Package dotnet-hostfxr-2.1.x86_64 0:2.1.0-1 will be installed
--> Processing Dependency: dotnet-host >= 2.1.0 for package: dotnet-hostfxr-2.1-2.1.0-1.x86_64
---> Package dotnet-runtime-deps-2.1.x86_64 0:2.1.0-1 will be installed
--> Processing Dependency: libicu for package: dotnet-runtime-deps-2.1-2.1.0-1.x86_64
--> Running transaction check
---> Package dotnet-host.x86_64 0:2.1.0_rc1-1 will be installed
--> Processing Dependency: dotnet-runtime-deps-2.1.0-rc1 >= 2.1.0-rc1 for package: dotnet-host-2.1.0_rc1-1.x86_64
---> Package libicu.x86_64 0:50.1.2-15.el7 will be installed
--> Running transaction check
---> Package dotnet-runtime-deps-2.1.0-rc1.x86_64 0:2.1.0_rc1-1 will be installed
--> Finished Dependency Resolution

Dependencies Resolved

================================================================================
 Package                 Arch   Version       Repository                   Size
================================================================================
Installing:
 dotnet-sdk-2.1          x86_64 2.1.300-1     /dotnet-sdk-2.1.300-x64     219 M
Installing for dependencies:
 aspnetcore-runtime-2.1  x86_64 2.1.0-1       packages-microsoft-com-prod  29 M
 dotnet-host             x86_64 2.1.0_rc1-1   packages-microsoft-com-prod  43 k
 dotnet-hostfxr-2.1      x86_64 2.1.0-1       packages-microsoft-com-prod 195 k
 dotnet-runtime-2.1      x86_64 2.1.0-1       packages-microsoft-com-prod  27 M
 dotnet-runtime-deps-2.1 x86_64 2.1.0-1       packages-microsoft-com-prod 2.8 k
 dotnet-runtime-deps-2.1.0-rc1
                         x86_64 2.1.0_rc1-1   packages-microsoft-com-prod 2.8 k
 libicu                  x86_64 50.1.2-15.el7 base                        6.9 M

Transaction Summary
================================================================================
Install  1 Package (+7 Dependent packages)
```

**Comment by stephenetyler** [comment]
I'm getting the same error downloading a previous release using
sudo yum install dotnet-runtime-2.0.7

Can someone fix this please?  and maybe a test should be written for this...

**Comment by johndesp** [comment]
I am getting the same issue as well, with any flavor of 2.1.x  

thanks

**Comment by gnjack** [comment]
Also having the same issue with missing `dotnet-host-2.1.0_rc1` and `dotnet-runtime-deps-2.1.0-rc1` packages.

You can work around it by explicitly installing version `2.1.0-1` of `dotnet-host` before installing the main sdk / runtime package. For example `yum install dotnet-host-2.1.0-1 dotnet-sdk-2.1`

Probably related to work being done in #1620 :
> We're in the process of removing the pre-release versions of dotnet-host-2.1.0 from the product feeds and this issue will be closed once that is complete.

**Comment by leecow** [comment]
@mrichards3 - yes, the RHEL release is still being finalized. Talked with the Red Hat folks yesterday and they're on track to release next week.

**Comment by tmds** [comment]
The RHEL rpms are available, you can install and use .NET Core 2.1 as follows:
```
$ sudo yum install rh-dotnet21 -y
$ scl enable rh-dotnet21 bash
```
RHEL container images (and documentation) are expected in 1 to 2 weeks.

**Comment by leecow** [comment]
Closing this issue now that 2.1.1 is released.
